src/main/java/kitchenpos/domain/menu/Menu.java

```diff
@@ -0,0 +1,111 @@
+package kitchenpos.domain.menu;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Objects;
+import javax.persistence.Column;
+import javax.persistence.Embedded;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.ForeignKey;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import kitchenpos.common.exception.CommonErrorCode;
+import kitchenpos.common.exception.InvalidParameterException;
+
+@Entity
+public class Menu {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(name = "name", nullable = false)
+    private String name;
+
+    @Embedded
+    private Price price;
+
+    @ManyToOne(fetch = FetchType.EAGER)
+    @JoinColumn(name = "menu_group_id", foreignKey = @ForeignKey(name = "fk_menu_menu_group"))
+    private MenuGroup menuGroup;
+
+    @Embedded
+    private MenuProducts menuProducts;
+
+    protected Menu() {
+    }
+
+    private Menu(String name, Price price, MenuGroup menuGroup, MenuProducts menuProducts) {
+        validEmpty(name, menuGroup);
+        validPrice(price, menuProducts.getSumPrice());
+        this.name = name;
+        this.menuGroup = menuGroup;
+        this.price = price;
+        this.menuProducts = menuProducts.mapMenu(this);
+    }
+
+    public static Menu of(String name, int price, MenuGroup menuGroup,
+        List<MenuProduct> menuProducts) {
+        return new Menu(name, Price.of(BigDecimal.valueOf(price)), menuGroup,
+            MenuProducts.of(menuProducts));
+    }
+
+    public Long getId() {
+        return id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public BigDecimal getPrice() {
+        return price.value();
+    }
+
+    public List<MenuProduct> getMenuProducts() {
+        return menuProducts.getMenuProducts();
+    }
+
+    public Long getMenuGroupId() {
+        return menuGroup.getId();
+    }
+
+    private void validPrice(Price price, BigDecimal sumPrice) {
```

Comment:
```suggestion
    private void validatePrice(Price price, BigDecimal sumPrice) {
```
메서드명은 동사구로 시작하는게 좋을 것 같습니다 :)
전체적으로 공통적인 확인 사항입니다.

---

클래스 내 인스턴스 메서드의 위치는 접근제어자로 정하신 걸까요 ?? ㅎ

Reply:
네 `private`은 아래에 두고 `equals, hashCode, toString` 보다는 위에 두도록 했습니다